draft-vvv-webtransport-http3.md

```diff
@@ -157,10 +157,61 @@ is closed.
 
 ## Establishing a Transport-Capable HTTP/3 Connection
 
+In WebTransport, an *exclusive mode* is a state of HTTP/3 connection where the
+connection is expected to have exactly one WebTransport session, and no other
+HTTP requests.
+
 In order to indicate support for WebTransport, both the client and the server
-MUST send a SETTINGS_ENABLE_WEBTRANSPORT value set to "1" in their SETTINGS
-frame.  Endpoints MUST NOT use any WebTransport-related functionality unless
-the parameter has been negotiated.
+MUST send a SETTINGS_ENABLE_WEBTRANSPORT setting.  The default value of
+SETTINGS_ENABLE_WEBTRANSPORT is "0" (Unsupported), however it can also assume
+values of "1" (Supported) and "2" (Exclusive).  The exact meaning of the value
+depends on the party.
+
+SETTINGS_ENABLE_WEBTRANSPORT sent by the client has the following possible values:
+* Unsupported ("0") indicates that the client does not support WebTransport
+  over HTTP/3.
+* Supported ("1") indicates that the client supports WebTransport over HTTP/3
+  and may potentially initiate a WebTransport session in the future.
+* Exclusive ("2") indicates that the client both supports WebTransport over
+  HTTP/3 and has a WebTransport request that it intends to send over the
+  connection in question.
+
+SETTINGS_ENABLE_WEBTRANSPORT sent by the server has the following possible values:
+* Unsupported ("0") indicates that the server does not support WebTransport
+  over HTTP/3.
+* Supported ("1") indicates that the server supports WebTransport over HTTP/3.
+* Exclusive ("2") indicates that the server expects to handle exactly one
+  request over the connection in question, and that request has to be a
+  WebTransport session.
+
+Once an HTTP/3 implementation has the SETTINGS_ENABLE_WEBTRANSPORT from both
+sides, the following should happen:
+
+* If one of client's or server's value is "0", and another value is either "0"
+  or "1", the connection does not support WebTransport.
+* If both server's value is "1", and client's value is "1" or "2", the
+  connection supports WebTransport in normal mode.
+* If server's value is "2", and client's value is "0" or "1", the connection
+  MUST be closed with a connection error.
```

Comment:
Echoing @DavidSchinazi's earlier point -- this makes it challenging to run a web server and an exclusive-only web transport server on the same address/port.  The server can send SETTINGS in 0.5-RTT, and it won't know if the client is going to send 0 (HTTP) or 2 (WT).  Alternately, the server can wait to receive the client SETTINGS before sending its own, but that's suboptimal for other reasons.

Reply:
Somewhat related is that there is a completely incompatible set of modes here -- C=1, S=2.  These clients can't talk at all, even if the client would prefer pooled connections but would settle for exclusive ones.

Reply:
I'm confused.  I don't think you can run a non-exclusive server and and exclusive server on the same port without multiplexing by something like ALPN.

If the client prefers pooled version, it can send C=1.  C=1, S=1 results in pooled WebTransport, C=1, S=2 results in exclusive WebTransport.

Reply:
Per the text and the table, C=1, S=2 is an error.